R/compare.R

```diff
@@ -56,9 +56,43 @@
 #' @param ignore_encoding Ignore string encoding? `TRUE` by default, because
 #'   this is R's default behaviour. Use `FALSE` when specifically concerned
 #'   with the encoding, not just the value of the string.
+#' @param ignore_name_order Ignore changes to the order of named components in
+#'   lists or vectors.
+#' @param ignore_NULLs In lists and pairlists, ignore elements that are set to
+#'   `NULL`.
+#' @param ignore_private Components matching the patterns in this vector are
+#'   completely ignored in comparisons.
 #' @returns A character vector with class "waldo_compare". If there are no
 #'   differences it will have length 0; otherwise each element contains the
 #'   description of a single difference.
+#'
+#' @details Objects may contain an attribute named `"waldo_opts"`, which should
+#'   be a list containing compare options.  If `x` or `y` has its own
+#'   components, each of those may contain a `"waldo_opts"` attribute.  If
+#'   present the options will be applied in the following order from lowest to
+#'   highest precedence:
+#'
+#'   1.  Defaults from this function.
+#'   2.  The `waldo_opts` for `x`.
+#'   3.  The `waldo_opts` for `y`.
+#'   4.  The `waldo_opts` for components of `x`.
+#'   5.  The `waldo_opts` for corresponding components of `y`.
+#'   6.  Continue recursively...
+#'   7.  User-specified arguments to `compare()` override
+#'       everything else.
+#'
+#'   By default the `"waldo_opts"` attribute is listed in
+#'   `ignore_attr` so that changes to it are not reported; if you
+#'   customize `ignore_attr`, you will probably want to do this
+#'   yourself.
+#'
+#'   The patterns listed in the `ignore_private` vector are
```

Comment:
I think this paragraph probably belongs in the `ignore_private` docs?

Reply:
Sure.  (The style I use usually has shorter param descriptions than you are using.)